lib/libsmb2.c

```diff
@@ -84,10 +84,18 @@
 #include <sys/socket.h>
 #endif
 
-#if defined(_WIN32) || defined(_XBOX)
+#if defined(_WIN32) || defined(_XBOX) || defined(__AROS__)
 #include "asprintf.h"
 #endif
 
+#if !defined(__amigaos4__) && (defined(__AMIGA__) || defined(__AROS__))
+#include <proto/bsdsocket.h>
+#undef getaddrinfo
+#undef freeaddrinfo
+#define close CloseSocket
+#endif
+
```

Comment:
move to compat.h ?

Reply:
moved here: https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa